tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/ui/pin/pinview/CalculatorKeyView.kt

```diff
@@ -6,49 +6,145 @@ package com.hzontal.tella_locking_ui.ui.pin.pinview
         if (resultListener != null) {
             mResultListener = resultListener
         }
-        mOnKeyBoardClickListener = OnKeyBoardClickListener(minPinLength,pinLockListener,this)
+        mOnKeyBoardClickListener = OnKeyBoardClickListener(minPinLength, pinLockListener, this)
         initListeners()
     }
 
-    private fun initListeners () {
-        mGroupButtons.referencedIds.forEach {
+    fun initTheme(style: CalculatorThemeStyle, calculatorTheme: CalculatorTheme) {
+
+        when (calculatorTheme) {
+            CalculatorTheme.BLUE_SKIN -> {
+                binding.btnOperatorsGroup.referencedIds.forEach {
+                    val button = findViewById<TextView>(it)
+                    button.setBackgroundColor(ContextCompat.getColor(context, style.calculatorOperatorsBackgroundColor))
+                    button.setTextColor(ContextCompat.getColor(context, style.calculatorOperatorsTextColor))
+                }
+
+                binding.btnNumbersGroup.referencedIds.forEach {
+                    val button = findViewById<TextView>(it)
+                    button.setBackgroundColor(ContextCompat.getColor(context, style.calculatorNumbersBackgroundColor))
+                    button.setTextColor(ContextCompat.getColor(context, style.calculatorNumbersTextColor))
+                }
+
+                binding.plusMinusBottomBtn.setBackgroundColor(ContextCompat.getColor(context, style.calculatorNumbersBackgroundColor))
+                binding.plusMinusBottomBtn.setTextColor(ContextCompat.getColor(context, style.calculatorOperatorsTextColor))
+                binding.commaBtn.setBackgroundColor(ContextCompat.getColor(context, style.calculatorNumbersBackgroundColor))
+                binding.commaBtn.setTextColor(ContextCompat.getColor(context, style.calculatorOperatorsTextColor))
+                binding.okBtn.setBackgroundColor(ContextCompat.getColor(context, style.calculatorOkBtnBackgroundColor))
+
+                binding.plusMinusBtn.text = "()"
+                binding.plusMinusBottomBtn.visibility = VISIBLE
+                binding.zeroBtnGuideline.setGuidelinePercent(0.25F)
+
+            }
+
+            CalculatorTheme.ORANGE_SKIN -> {
+
+                binding.btnsGroup.referencedIds.forEach {
+                    val button = findViewById<TextView>(it)
+                    button.setBackgroundResource(style.calculatorBackgroundDrawable)
+                    button.setTextColor(ContextCompat.getColor(context, style.calculatorNumbersTextColor))
+                }
+
+                binding.btnOperatorsGroup.referencedIds.forEach {
+                    val button = findViewById<TextView>(it)
+                    button.setBackgroundResource(style.calculatorBackgroundDrawable)
+                    button.setTextColor(ContextCompat.getColor(context, style.calculatorOperatorsTextColor))
+                }
+                binding.okBtn.setBackgroundResource(style.calculatorOkBtnTextBackgroundDrawable)
```

Comment:
You can use a scope function to apply the style you need like :

`button.apply {
setBackgroundResource(style.calculatorBackgroundDrawable)
}`. and let's apply this to the rest of the code please.